app/models/action/edition_diff_factory.rb

```diff
@@ -0,0 +1,48 @@
+class Action::EditionDiffFactory
+  def add_previous_edition(previous_edition)
+    @previous_edition = get_edition_attributes(previous_edition)
+  end
+
+  def add_updated_edition(updated_edition)
+    @updated_edition = get_edition_attributes(updated_edition)
+  end
+
+  def diff
+    raise MissingEditionError, 'No previous_edition provided' if @previous_edition.nil?
+    raise MissingEditionError, 'No updated_edition provided' unless @updated_edition.present?
+    create_edition_diff(@previous_edition, @updated_edition)
+  end
+
+private
+
+  def create_edition_diff(edition_a, edition_b)
+    HashDiff.diff(edition_a, edition_b)
+  end
+
+  def get_edition_attributes(edition)
+    attributes = {}
+    return attributes unless edition.present?
+
+    Edition::TOP_LEVEL_FIELDS.each do |field|
+      attributes[field.to_s] = edition.send(field)
```

Comment:
Minor thing, but you're probably better to use `public_send` here as that won't accidentally send a private method. It's not really likely in this scenario, but good to indicate to future readers that you're only using public methods.